lib/dec/avx2.c

```diff
@@ -36,11 +36,11 @@ while (srclen >= 45)
 
 	/* Check if all bytes have been classified; else fall back on bytewise code
 	 * to do error checking and reporting: */
-	if ((unsigned int)_mm256_movemask_epi8(s1mask | s2mask | s3mask | s4mask | s5mask) != 0xFFFFFFFF) {
+	if ((unsigned int)_mm256_movemask_epi8(_mm256_or_si256(_mm256_or_si256(_mm256_or_si256(_mm256_or_si256(s1mask, s2mask), s3mask), s4mask), s5mask)) != 0xFFFFFFFF) {
```

Comment:
I will probably make the change above (using temporary variables) this weekend as well. Its time has come.